### ISSUE TITLE
Empty request slots crash parser

### ISSUE THREAD

**Issue by xiagu** [Teoxoy/factorio-blueprint-editor#51]
#### TL;DR
Requester chests edited by hand get a 12-length request filter array regardless of how many slots are actually set. When parsing these empty slots, the parser expects them to have a `name` property, but they do not.

#### Steps to reproduce
1. Place a logistic requester chest.
2. Edit the filter and request something, like transport belts.
3. Copy the blueprint string.
4. Paste the blueprint string.

#### Expected
Blueprint string should be parsed and reloaded.

#### Actual
Parsing fails with an error logged to the console:
```
0: {keyword: "required", dataPath: ".blueprint.entities[0].request_filters[1]", schemaPath: "#/properties/entities/items/properties/request_filters/items/required", params: {…}, message: "should have required property 'name'"}
1: {keyword: "additionalProperties", dataPath: "", schemaPath: "#/oneOf/1/additionalProperties", params: {…}, message: "should NOT have additional properties"}
2: {keyword: "oneOf", dataPath: "", schemaPath: "#/oneOf", params: {…}, message: "should match exactly one schema in oneOf"}
```

Simple repro:
https://teoxoy.github.io/factorio-blueprint-editor/?source=0eJyFUdFuwyAM/Bc/Eylka5PyuN+YqiqhXmeJQApkWhTl3+s0XeS3vcAZznc+mKFzIw6RfAYzA9ngE5jPGRLdfOvWszwNCAYoYw8KfNuvlQs3SplsYb8x5SLifeQdIywKyF/xF4xezgrQZ8qEm+Z/vQqGkJge/OrLEqWCiVfWvFJEu92UL9Xp4se+4zajFbxELl/kWGmz+5tjHzrH1qchxFx06DLb2TCuuXXJFju/EvhN4HeBDwIfBa4FbgQ+CaylmdayqJbz+mjPpzbiZxT8cKhn+mNTV1WjT2VdKXAtB2Hmx85clgdjypxR
(will crash and not render)

**Comment by Teoxoy** [comment]
Thanks for the report! I will take a look in a bit.
